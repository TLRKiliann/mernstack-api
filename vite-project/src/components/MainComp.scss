.master--maincomp {
	margin-top: 40px;
	width: 40%;
	margin: auto;
	height: auto;
}

.submaster--maincomp {
}

.bg--colormaincomp {
	width: 100%;
	margin-top: 40px;
	padding: 20px 0px;
 	border: 2mm ridge royalblue;
 	border-radius: 12px;
 	box-shadow: 0px 0px 4px whitesmoke;
	background-color: rgba(45, 85, 255, 0.4);
}

.h2--maincomp {
	text-align: center;
	font-family: 'Consolas';
	text-shadow: 2px 2px 2px blue; 
	color: lightblue;
}

.div--connection {
	width: 80%;
	padding: 5px 10px;
	margin: 10px auto;
	display: flex;
	align-items: center;
	background-color: lightblue;
 	border: 2mm ridge royalblue;
 	border-radius: 7px;
 	box-shadow: 0px 0px 4px whitesmoke;
	color: royalblue;
}