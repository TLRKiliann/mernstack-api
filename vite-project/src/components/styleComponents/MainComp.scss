$pink: #ffc0cb;
$purple: rgba(195,87,116, 1);
$lightgreen: #90ee90;
$skyblue: #50bcfe;
$font_acme: 'Acme', sans-serif;
$font_dejavu: 'DejaVu Sans', sans-serif;
$font_junction: 'Junction', sans-serif;

@mixin margin-font-family-size($size){
  margin: auto;
  text-align: center;
  font-family: $font_acme;
  font-size: $size;
  color: rgba(244,164,96, 1);
}

@mixin family-size-color($font, $sizef, $color){
  font-family: $font;
  font-size: $sizef;
  color: $color;
}

@mixin family-size($family, $sizefont){
  font-family: $family;
  font-size: $sizefont;
}

.master--maincomp {
  width: 500px;
  height: auto;
  margin: auto;
  margin-top: 20px;
  margin-bottom: 140px;
  .bg--colormaincomp {
    margin: auto;
    padding-top: 20px;
    padding-bottom: 40px;
    background: linear-gradient(90deg,
      rgba(244,164,96, 0.5) -20%,
      rgba(195,87,116, 0.5) 100%);
    border: none;
    outline: none;
    border-radius: 12px;
    box-shadow: 0px 0px 20px rgba(195,87,116, 0.2);
    .h2--maincomp {
      @include margin-font-family-size(2.4rem);
      text-shadow: 2px 1px 1px $purple;
      .span--lengthuser {
        text-shadow: 2px 1px 1px $purple;
        color: $pink;
      }
    }
    .h3--maincomp {
      @include margin-font-family-size(1.8rem);
      text-shadow: 2px 1px 1px $purple;
    }
  }
  .membershome__div {
    width: 80%;
    height: 60px;
    margin: auto;
    margin-top: 10px;
    display: flex;
    align-items: center;
    @include family-size-color($font_dejavu, 1.1rem, $skyblue);
    background-color: rgba(85, 45, 255, 0.4);
    border-radius: 7px;
    .membershome__divimg {
      position: absolute;
      width: 44px;
      height: 44px;
      margin: auto;
      margin-top: 5px;
      margin-right: 5px;
      margin-bottom: 5px;
      margin-left: 5px;
      border-radius: 50%;
      .membershome__img {
        width: 100%;
        height: 100%;
        margin: auto;
        display: flex;
        object-fit: cover;
        border-radius: 50%;
      }
    }
    .membershome__names {
      margin: auto;
      margin-left: 60px;
      display: flex;
    }
    .membershome__connection {
      margin: auto;
      margin-right: 10px;
      display: flex;
      @include family-size-color($font_junction, 1.2rem, $lightgreen);
    }
  }
}

@media only screen and (min-width: 360px) and (max-width: 1050px) {
  .master--maincomp {
    width: 80%;
    height: auto;
    margin: auto;
    margin-top: 80px;
    margin-bottom: 80px;
    .bg--colormaincomp {
      .h2--maincomp {
        font-size: 1.4rem;
        .span--lengthuser {
        }
      }
      .h3--maincomp {
        @include family-size($font_acme, 1.2rem);
      }
    }
    .membershome__div {
      @include family-size($font_dejavu, 1.0rem);
      .membershome__divimg {
        .membershome__img {
        }
      }
      .membershome__names {
        @include family-size($font_dejavu, 0.9rem);
      }
      .membershome__connection {
        @include family-size($font_dejavu, 0.9rem);
      }
    }
  }
}