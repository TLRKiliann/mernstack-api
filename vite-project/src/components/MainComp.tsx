import React from 'react'
import { userType } from '../models/userType';
import './styleComponents/MainComp.scss'

interface MainCompProps {
  users: userType[]
}

const MainComp: React.FC = (props: MainCompProps) => {
  return(
    <div className="master--maincomp">
          
      <div className="bg--colormaincomp">

        <h2 className='h2--maincomp'>
          All Members :&nbsp;
          <span style={{color: "orange"}}>
            {props.users?.length}
          </span>
        </h2>
        
        <h3 className="h3--maincomp">
          Users online (connected) :&nbsp;
          <span
            style={{color:'whitesmoke'}}
          >
            {props.users?.reduce((prev, curr) =>
            prev + curr.isConnected, 0)}
          </span>
        </h3>

        <div>
          {props.users?.slice(0, `${props.users?.length}`).map(user => (
            <div key={user.id}>
            {user.isConnected ? (
              <div className="div--connection">
                <img
                  src={user.img}
                  width="40px"
                  height="40px"
                  className="img--connector"
                  alt={user.img}
                />
                {user.firstName} {user.lastName} :&nbsp;
                  <div 
                    style={{
                      fontFamily: 'Junction',
                      fontSize: '1.2rem',
                      color: 'lightgreen'
                    }}>
                    Connected
                  </div>
              </div>
              ) : null
            }
            </div>
            ))
          }
        </div>
      </div>
    </div>
  )
}

export default MainComp