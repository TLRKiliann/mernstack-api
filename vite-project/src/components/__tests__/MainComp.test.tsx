import React from "react";
import { screen, render } from '@testing-library/react';
import { expect, vi } from 'vitest'
import { create } from 'react-test-renderer'
import '@testing-library/jest-dom'
import MainComp from "../MainComp.tsx";

test('MainComp get by text', () => {
  render(<MainComp />);
  const linkElement = screen.getByText(/all members/i);
  expect(linkElement).toBeInTheDocument();
})

test('MatchSnapShot test MainComp', () => {
  const treeMainComp = create(<MainComp />)
  expect(treeMainComp.toJSON()).toMatchSnapshot()
})