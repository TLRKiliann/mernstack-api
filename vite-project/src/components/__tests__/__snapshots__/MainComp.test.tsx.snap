// Vitest Snapshot v1

exports[`MatchSnapShot test MainComp 1`] = `
<div
  className="master--maincomp"
>
  <div
    className="bg--colormaincomp"
  >
    <h2
      className="h2--maincomp"
    >
      All Members : 
      <span
        className="span--lengthuser"
      />
    </h2>
    <h3
      className="h3--maincomp"
    >
      Users online (connected) : 
      <span
        style={
          {
            "color": "pink",
          }
        }
      />
    </h3>
    <div />
  </div>
</div>
`;
